code/modules/hydroponics/plants_flower.dm

```diff
@@ -20,3 +20,36 @@ ABSTRACT_TYPE(/datum/plant/flower)
 	force_seed_on_harvest = 1
 	mutations = list()
 	commuts = list(/datum/plant_gene_strain/immunity_radiation,/datum/plant_gene_strain/damage_res/bad)
+
+/datum/plant/flower/rafflesia
+	name = "Rafflesia"
+	seedcolor = "#A4000F"
+	crop = /obj/item/clothing/head/rafflesia
+	starthealth = 40
+	growtime = 50
+	harvtime = 90
+	cropsize = 1
+	harvests = 1
+	isgrass = 1
+	endurance = 5
+	genome = 9
+	force_seed_on_harvest = 1
+	special_proc = 1
+	mutations = list()
+	commuts = list(/datum/plant_gene_strain/resistance_drought)
+	assoc_reagents = list("miasma")
+	
+	HYPspecial_proc(var/obj/machinery/plantpot/POT) // Smokes miasma and whatever chemicals have been spliced into the plant
+		..()
+		if (.) return
+		var/datum/plant/P = POT.current
+		var/datum/plantgenes/DNA = POT.plantgenes	
+		var/spray_prob = max(30,(30 + DNA.endurance / 10))
+	
+		if (POT.growth > (P.harvtime - DNA.growtime) && prob(spray_prob))
+			POT.reagents.clear_reagents() // Prevents smoking anything you spill into the pot,  that and pottasium water explosions.
+			for(var/REAG in assoc_reagents)
+				POT.reagents.add_reagent(REAG, max(1,(1 + DNA.potency / 5)))
```

Comment:
Would rather this *only* smoke miasma. Hotboxing is an issue often enough without another way for massive smokes to happen

Reply:
Yeah that is a valid point. Though if the plant would only smoke miasma then it would be a pointless update. I think that a better way of doing would be either to limit how much smoke can be produced or to create several mutations with different chemicals, though im more open to reducing the ammount of smoke, since i want to leave lots of options with it.

Reply:
This smoking is heavily limited by both endurance and potency, similar how balloon melons flooding effect is. I honestly don't see the problem with it being robust all-smoke plant, mainly because is HYPspecial is bugged and wont be fixed anytime soon.